### ISSUE TITLE
 withPageAuthRequired unauthorised flashes failed to load script error

### ISSUE THREAD

**Issue by esharmony** [auth0/nextjs-auth0#603]
This issues is when I use the auth0 method:  **withPageAuthRequired**
When I am not logged in and I try to access a protected page I get the failed to load script flashing before the redirect takes place.

My page code below: 

`
import { CardContent } from '@mui/material';
import { useRouter } from 'next/router';
import Layout from '../components/layout';
import type { ReactElement } from 'react';
import { get } from '../utils/http';
import { withPageAuthRequired } from '@auth0/nextjs-auth0';

interface AboutProps {
  data: string;
}

interface Data {
  data: string;
}

function About(props: AboutProps) {
  const router = useRouter();
  const { locale, locales } = router;

  return (
    <CardContent>
      {/* <Results /> */}Here is the about content with {locale}
      <p>Configured locales: {JSON.stringify(locales)}</p>
      <p>{props.data}</p>
    </CardContent>
  );
}

About.getLayout = function getLayout(page: ReactElement) {
  return <Layout>{page}</Layout>;
};

export const getServerSideProps = withPageAuthRequired({
  async getServerSideProps(ctx) {
    return {
      props: {
        messages: (await import(`../messages/${ctx.locale}.json`)).default,
        data: await get<Data>('http://localhost:3000/api/hello', {
          headers: { Cookie: ctx.req.headers.cookie! },
        }).then((user) => user.parsedBody?.data),
      },
    };
  },
});

export default About;

`

everything works as expected when I am logged in, though I could only use the withApiAuthRequired when passing the Cookie data as shown.

**Comment by esharmony** [comment]
@adamjmcgrath Thanks I understand what you are saying for the API, 

I will just create exportable functions that will connect to the DB and import these functions for the Next app.
For my mobile app endpoints I will call those functions also from the default endpoint which will be protected by validating the Auth0 token.

This strategy seems fine to me (but maybe not?)

I am still wondering though if what I have done for unauthorised page redirect is optimal.

Because of internationalisation I have overwritten the handleLogin, this accepts a parameter that is picked up by auth0 and switches my language.

So when a user hits a protected page (logged out) I have overwritten the redirect to a new page using 

```
// .env.local 
...
NEXT_PUBLIC_AUTH0_LOGIN=/loginRedirect  
```

and then 

```
// /pages/loginRedirect.tsx 

import { useRouter } from 'next/router';
import React, { useEffect } from 'react';

function LoginRedirect() {
  const router = useRouter();
  useEffect(() => {
    window.location.replace(
      `/api/auth/login?lang=${router.locale || 'en'}&returnTo=${
        router.query.returnTo || '/'
      }`
    );
  }, [router]);
  return (
    <p
      style={{
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
      }}
    >
      redirecting... // need intl 
    </p>
  );
}

export default LoginRedirect;
```

So the user will see a page saying redirecting before they are redirected to login on auth0 
I don't see how I can do this on the server rather than loading this extra page allowing me to also retrieve the locale info?

Is there a better way? 

Many thanks

**Comment by adamjmcgrath** [comment]
> This strategy seems fine to me (but maybe not?)

Yep - this is what [vercel recommends](https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#getserversideprops-or-api-routes)

> I don't see how I can do this on the server rather than loading this extra page allowing me to also retrieve the locale info?

Can you not get the locale from the protected page and call `withPageAuthRequired` with the internationalized returnTo url, like the `my-ssr-page.jsx` [example on the readme ](https://github.com/auth0/nextjs-auth0#base-path-and-internationalized-routing)?

**Comment by esharmony** [comment]
In my mind the returnTo parameter cannot effect the actual modal language, not that I found anyway.

**Comment by adamjmcgrath** [comment]
> I think also maybe the redirect does not like to call this as its in the pages/api/auth folder, maybe this is wrong?

I'm not entirely sure what you mean, but I can't see anything wrong with the code you've shared

**Comment by esharmony** [comment]
@adamjmcgrath I have done a deep dive and also created the most basic repo i could to show what I am going to do.

https://github.com/esharmony/intl-next-auth0-xmpl

Basically the only way I have been able to achieve (being able to switch the modal language)  is to pick up the language from the returnUrl parameter in the api/auth/login file which I overwrote.
You can see how i set the locale as you suggested on the protected page, but I have to split this url get the slug and check  is an accepted locale, this is a dodgy hack.

I could not find any other way of getting the locale info to the method so I could pass it along to the auth0 backend via the authorizationParams. 

Maybe I am being completely dumb and missing something simple, I have been down so overly complicate attempts already so it would not surprise me if I am being so : ) 

You can just check my protected page and then the api/auth/login file and it should be clear what I am doing.
